src/components/UserProfile.js

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Button,
+  Card,
+  CardContent,
+  CardActions,
+  Table,
+  TableRow,
+  TableCell,
+  getImageListItemBarUtilityClass,
+} from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+// import { getUserProfile } from './api';
+import { Avatar } from '@mui/material';
+import axios from 'axios';
+
+export default function PersonalProfile() {
+  const navigate = useNavigate();
+  const { user, status } = useAuth();
+  const [userData, setUserData] = useState(null); 
+  const [showCreditCardInfo, setShowCreditCardInfo] = useState(false);
+
+  
+  const handleSeeMoreClick = () => {
+    setShowCreditCardInfo(!showCreditCardInfo);
+  };
+
+  useEffect(() => {
+    if (status === AuthStatus.LoggedIn) {
+
+      axios.get('http://localhost:8000/api/v1/user') 
+      .then((response) => {
+          setUserData(response.data); // Update user data in state
+        })
+        .catch((error) => {
+          // Handle errors, e.g., display an error message
+          console.error('Error fetching user data:', error);
+        });
+    }
+  }, 
+  [status]);
```

Comment:
Since the `useAuth` hook already includes the user info, we don't need to make a second call to the backend here. But for future reference, there's a few things to point out:

*   instead of using axios directly, we've been importing a base axios instance from the `apis` folder here: [https://github.com/Code-the-Dream-School/dd-prac-team4-front/blob/FetchUserProfilegit/src/apis/axiosClient.js](https://github.com/Code-the-Dream-School/dd-prac-team4-front/blob/FetchUserProfilegit/src/apis/axiosClient.js)
    *   The reason we've been doing this is because we've set some default values there such as the Base URL, so that the code knows to make the backend request to `localhost:8000` when running in development but instead to [`https://musicstore.onrender.com`](https://musicstore.onrender.com) when running in production. It walso handles setting up cookies and any other headers that we may need
*   The backend API docs are all available on [https://musicstore.onrender.com/api-docs/](https://musicstore.onrender.com/api-docs/) . You'll see there that there is no endpoint `http://localhost:8000/api/v1/user` which is likely why the API call was failing when you were testing. Rather it's the plural form: `http://localhost:8000/api/v1/users` but that would give you _all users_ and is only available for admins. Instead, in this case we want to get info about the _current user_ so we woulduse `http://localhost:8000/api/v1/users/showMe` . 

But since in this case we don't actually need to make this API call, you can delete the whole `useEffect` block altogether.